datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -69,7 +70,11 @@ where
                     }).map_err(SyncFailure::new)
                     .from_err()
             },
-        )
+        ).map_err(move |e| {
+            e.into_inner().unwrap_or_else(move || {
+                format_err!("Ethereum node took too long to respond to eth_getLogs (from = {}, to = {}, # of event sigs = {})", from, to, event_sig_count)
```

Comment:
This line is a _bit_ too long. Wrap the string around with e.g.
```rust
"Ethereum node took too long to respond to eth_getLogs \
 (from block {}, to block {}, {} event signatures)"
```

Also, note: I've changed the error message a little. I think it is a little easier to read this way.